From f6cd4a67d7de5d3ff1a6e58a8c83749fc8ffc310 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michel=20D=C3=A4nzer?= <michel.daenzer@amd.com>
Date: Fri, 8 Mar 2019 11:48:10 +0100
Subject: [PATCH] Revert "glamor: Avoid glamor_create_pixmap for pixmaps
 backing windows"

This reverts commit 274703087f80342f51fa69c935bb9a1cb0c4ae47.

Reports of visual corruption were bisected to this, e.g.
https://bugs.archlinux.org/task/61941 . I can reproduce this with Turks,
but not with Bonaire. I assume it's a Mesa/glamor bug, but let's revert
for now.

Acked-by: Alex Deucher <alexander.deucher@amd.com>
---
 src/radeon_glamor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/radeon_glamor.c b/src/radeon_glamor.c
index 3e676f2d..f1098381 100644
--- a/src/radeon_glamor.c
+++ b/src/radeon_glamor.c
@@ -238,7 +238,7 @@ radeon_glamor_create_pixmap(ScreenPtr screen, int w, int h, int depth,
 		if (info->shadow_primary) {
 			if (usage != CREATE_PIXMAP_USAGE_BACKING_PIXMAP)
 				return fbCreatePixmap(screen, w, h, depth, usage);
-		} else if (usage != CREATE_PIXMAP_USAGE_BACKING_PIXMAP) {
+		} else {
 			pixmap = glamor_create_pixmap(screen, w, h, depth, usage);
 			if (pixmap)
 			    return pixmap;
-- 
2.18.1

